core/services/kraken/kraken.py

```diff
@@ -35,6 +36,9 @@ def client(self) -> aiodocker.Docker:
     async def run(self) -> None:
         while self.should_run:
             await asyncio.sleep(5)
+            if self.deleting_in_progress:
+                continue
+
```

Comment:
Could this lead to a loophole where kraken gets stuck in?

Reply:
logically, no, the `remove` operation ends setting `deleting_in_progress` as `False`.

Reply:
As long as it doesn't raise, yup.